setuptools/msvc9_support.py

```diff
@@ -57,6 +58,10 @@ def patch_for_specialized_compiler():
     Microsoft Visual C++ 14.0:
         Microsoft Visual C++ Build Tools 2015 (x86, x64, arm)
     """
+    if platform.system() != 'Windows':
```

Comment:
This check feels a bit redundant, or maybe an over-optimization. What's the motivation behind it? It seems to me if a platform has the relevant distutils module, it should be patched.